### ISSUE TITLE
CI: K8sServicesTest Checks service across nodes Tests NodePort

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#7125]
Output:

```
Stacktrace

/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:408
Pod "testclient-vm85q" can not connect to service "http://10.96.174.249:10080"
Expected command: kubectl exec -n default testclient-vm85q -- curl -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 http://10.96.174.249:10080 -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 time-> DNS: '0.000017()', Connect: '0.000000',Transfer '0.000000', total '5.001948'
Stderr:
 	 

/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/k8sT/Services.go:184
```

Link: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2381/testReport/junit/k8s-1/8/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort/

Logs: 
[ecef3454_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort.zip](https://github.com/cilium/cilium/files/2877149/ecef3454_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort.zip)

**Comment by ianvernon** [comment]
Another failure:

Link: https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2391/testReport/junit/k8s-1/13/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort/

Output: 
```
Stacktrace

/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:409
Pod "testclient-4f475" can not connect to service "http://10.98.101.203:10080"
Expected command: kubectl exec -n default testclient-4f475 -- curl -s -D /dev/stderr --fail --connect-timeout 5 --max-time 8 http://10.98.101.203:10080 -w "time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}',Transfer '%{time_starttransfer}', total '%{time_total}'" 
To succeed, but it failed:
Exitcode: 28 
Stdout:
 	 time-> DNS: '0.000021()', Connect: '0.000000',Transfer '0.000000', total '5.001181'
Stderr:
 	 command terminated with exit code 28
	 

/home/jenkins/workspace/cilium-ginkgo_cilium_master/src/github.com/cilium/cilium/test/k8sT/Services.go:184
Standard Output

Number of 'context deadline exceeded' in logs: 0
Number of 'level=error' in logs: 0
Number of 'level=warning' in logs: 4
Number of 'Cilium API handler panicked' in logs: 0
Number of 'Goroutine took lock for more than' in logs: 0
Cilium pods: [cilium-4dq52 cilium-vrnr4]
Netpols loaded: 
CiliumNetworkPolicies loaded: 
Endpoint Policy Enforcement:
Pod                              Ingress   Egress
app2-5b49d5dd9-bgrnh             false     true
testclient-4f475                           
coredns-fff89c9b9-z6vdl          false     false
etcd-operator-5cf67779fd-chfzs   false     false
cilium-etcd-bpd2sk5xld           false     false
cilium-etcd-d8xx545llh           false     false
cilium-etcd-hfbdv578dj           false     false
app3-645549cb84-7cq68            false     false
testclient-wjnpq                           
testds-gqkmn                               
testds-ldmll                               
Cilium agent 'cilium-4dq52': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 45 Failed 0
Cilium agent 'cilium-vrnr4': Status: Ok  Health: Ok Nodes "k8s2 k8s1" ContinerRuntime: Ok Kubernetes: Ok KVstore: Ok Controllers: Total 33 Failed 0

Standard Error

STEP: Making ten HTTP requests from "testclient-4f475" to "http://10.98.101.203:10080"
=== Test Finished at 2019-02-15T21:25:50Z====
===================== TEST FAILED =====================
cmd: kubectl get pods -o wide --all-namespaces
Exitcode: 0 
Stdout:
 	 NAMESPACE     NAME                                   READY   STATUS    RESTARTS   AGE     IP              NODE   NOMINATED NODE   READINESS GATES
	 default       testclient-4f475                       1/1     Running   0          21s     10.10.0.178     k8s1   <none>           <none>
	 default       testclient-wjnpq                       1/1     Running   0          21s     10.10.1.234     k8s2   <none>           <none>
	 default       testds-gqkmn                           1/1     Running   0          21s     10.10.1.211     k8s2   <none>           <none>
	 default       testds-ldmll                           1/1     Running   0          21s     10.10.0.115     k8s1   <none>           <none>
	 kube-system   cilium-4dq52                           1/1     Running   0          68s     192.168.36.12   k8s2   <none>           <none>
	 kube-system   cilium-etcd-bpd2sk5xld                 1/1     Running   0          8m16s   10.10.1.173     k8s2   <none>           <none>
	 kube-system   cilium-etcd-d8xx545llh                 1/1     Running   0          8m57s   10.10.0.210     k8s1   <none>           <none>
	 kube-system   cilium-etcd-hfbdv578dj                 1/1     Running   0          9m37s   10.10.1.60      k8s2   <none>           <none>
	 kube-system   cilium-etcd-operator-d6cb86868-gj765   1/1     Running   0          10m     192.168.36.12   k8s2   <none>           <none>
	 kube-system   cilium-vrnr4                           1/1     Running   0          66s     192.168.36.11   k8s1   <none>           <none>
	 kube-system   coredns-fff89c9b9-z6vdl                1/1     Running   0          10m     10.10.1.121     k8s2   <none>           <none>
	 kube-system   etcd-k8s1                              1/1     Running   0          22m     192.168.36.11   k8s1   <none>           <none>
	 kube-system   etcd-operator-5cf67779fd-chfzs         1/1     Running   0          10m     10.10.1.192     k8s2   <none>           <none>
	 kube-system   kube-apiserver-k8s1                    1/1     Running   0          22m     192.168.36.11   k8s1   <none>           <none>
	 kube-system   kube-controller-manager-k8s1           1/1     Running   0          23m     192.168.36.11   k8s1   <none>           <none>
	 kube-system   kube-proxy-j4hx7                       1/1     Running   0          20m     192.168.36.12   k8s2   <none>           <none>
	 kube-system   kube-proxy-nnpjf                       1/1     Running   0          23m     192.168.36.11   k8s1   <none>           <none>
	 kube-system   kube-scheduler-k8s1                    1/1     Running   0          23m     192.168.36.11   k8s1   <none>           <none>
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-4dq52 -- cilium service list
Exitcode: 0 
Stdout:
 	 ID   Frontend              Backend                   
	 1    10.96.0.1:443         1 => 192.168.36.11:6443   
	 2    10.96.0.10:53         1 => 10.10.1.121:53       
	 3    10.104.138.92:2379    1 => 10.10.0.210:2379     
	                            2 => 10.10.1.173:2379     
	                            3 => 10.10.1.60:2379      
	 6    10.99.185.102:80      1 => 10.10.1.211:80       
	                            2 => 10.10.0.115:80       
	 7    10.98.101.203:10080   1 => 10.10.0.115:80       
	                            2 => 10.10.1.211:80       
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-4dq52 -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                              IPv6                   IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                                
	 706        Disabled           Disabled          4          reserved:health                                          f00d::a0a:100:0:8b57   10.10.1.96    ready   
	 1340       Disabled           Disabled          100        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:c03a   10.10.1.192   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=cilium-etcd-sa                                                
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 1365       Disabled           Disabled          104        k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:cf2d   10.10.1.121   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=coredns                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	                                                            k8s:k8s-app=kube-dns                                                                                  
	 1769       Disabled           Disabled          50652      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:f507   10.10.1.211   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
	                                                            k8s:zgroup=testDS                                                                                     
	 2805       Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:55e6   10.10.1.60    ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 4030       Disabled           Disabled          50628      k8s:io.cilium.k8s.policy.cluster=default                 f00d::a0a:100:0:3972   10.10.1.234   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.kubernetes.pod.namespace=default                                                               
	                                                            k8s:zgroup=testDSClient                                                                               
	 4091       Disabled           Disabled          101        k8s:app=etcd                                             f00d::a0a:100:0:f72    10.10.1.173   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                          
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                              
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                                       
	                                                            k8s:io.cilium/app=etcd-operator                                                                       
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                           
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-vrnr4 -- cilium service list
Exitcode: 0 
Stdout:
 	 ID   Frontend              Backend                   
	 1    10.96.0.10:53         1 => 10.10.1.121:53       
	 2    10.104.138.92:2379    1 => 10.10.1.60:2379      
	                            2 => 10.10.0.210:2379     
	                            3 => 10.10.1.173:2379     
	 3    10.96.0.1:443         1 => 192.168.36.11:6443   
	 6    10.99.185.102:80      1 => 10.10.0.115:80       
	                            2 => 10.10.1.211:80       
	 7    10.98.101.203:10080   1 => 10.10.0.115:80       
	                            2 => 10.10.1.211:80       
	 
Stderr:
 	 

cmd: kubectl exec -n kube-system cilium-vrnr4 -- cilium endpoint list
Exitcode: 0 
Stdout:
 	 ENDPOINT   POLICY (ingress)   POLICY (egress)   IDENTITY   LABELS (source:key[=value])                       IPv6                 IPv4          STATUS   
	            ENFORCEMENT        ENFORCEMENT                                                                                                       
	 1074       Disabled           Disabled          50628      k8s:io.cilium.k8s.policy.cluster=default          f00d::a0a:0:0:52eb   10.10.0.178   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
	                                                            k8s:zgroup=testDSClient                                                                      
	 1452       Disabled           Disabled          4          reserved:health                                   f00d::a0a:0:0:21b3   10.10.0.223   ready   
	 2312       Disabled           Disabled          101        k8s:app=etcd                                      f00d::a0a:0:0:16a7   10.10.0.210   ready   
	                                                            k8s:etcd_cluster=cilium-etcd                                                                 
	                                                            k8s:io.cilium.k8s.policy.cluster=default                                                     
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.cilium/app=etcd-operator                                                              
	                                                            k8s:io.kubernetes.pod.namespace=kube-system                                                  
	 2795       Disabled           Disabled          50652      k8s:io.cilium.k8s.policy.cluster=default          f00d::a0a:0:0:b420   10.10.0.115   ready   
	                                                            k8s:io.cilium.k8s.policy.serviceaccount=default                                              
	                                                            k8s:io.kubernetes.pod.namespace=default                                                      
	                                                            k8s:zgroup=testDS                                                                            
	 
Stderr:
 	 

===================== Existing AfterFailed =====================
```

Logs: 
[5ceabf58_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort.zip](https://github.com/cilium/cilium/files/2877191/5ceabf58_K8sServicesTest_Checks_service_across_nodes_Tests_NodePort.zip)

**Comment by brb** [comment]
One more: https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/5973/testReport/junit/k8s-1/8/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort/

**Comment by brb** [comment]
https://jenkins.cilium.io/job/cilium-master-K8s-all/166/testReport/junit/k8s-1/9/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort/

**Comment by jrfastab** [comment]
I'm also hitting this,

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/10184/testReport/junit/k8s-1/13/K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_service_on_same_node/

**Comment by jrajahalme** [comment]
https://jenkins.cilium.io/job/cilium-master-K8s-all/166/

**Comment by borkmann** [comment]
Hit from PR:

[c36538ea_K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_service_on_same_node (1).zip](https://github.com/cilium/cilium/files/3121376/c36538ea_K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_service_on_same_node.1.zip)

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/11900/testReport/junit/Suite-k8s-1/10/K8sServicesTest_Checks_ClusterIP_Connectivity_Checks_service_on_same_node/

Warnings/errors found in Cilium agent log:

```
2019-04-26T12:28:17.300045436Z level=debug msg="Queueing container event" containerID=a5900cdabc event="exec_start: /bin/sh -ec ETCDCTL_API=3 etcdctl --endpoints=127.0.0.1:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt --key=/etc/kubernetes/pki/etcd/healthcheck-client.key get foo" subsys=workload-watcher
2019-04-26T12:28:17.307146988Z level=error msg="Unable to parse Kafka request; closing Kafka request connection" error="unreasonable message/block size 1195725860 (max:6553500)" id="rx:10.10.0.74:55808->10.10.0.67:80<->tx:closed" subsys=proxy
2019-04-26T12:28:17.307356818Z level=debug msg="Closing socket" conn="rx:10.10.0.74:55808->10.10.0.67:80" subsys=proxy
2019-04-26T12:28:17.307550478Z level=debug msg="Rx is already closed, waiting for Tx to close" connPair="rx:10.10.0.74:55808->10.10.0.67:80<->tx:closed" subsys=proxy
```

```
2019-04-26T12:22:34.247847149Z level=debug msg="Adding CiliumNetworkPolicy" ciliumNetworkPolicyName=foo k8sApiVersion= k8sNamespace=default subsys=daemon
2019-04-26T12:22:34.247895636Z level=warning msg="Unable to add CiliumNetworkPolicy" ciliumNetworkPolicyName=foo error="Invalid CiliumNetworkPolicy specs: L7 rules can only apply to TCP (not UDP) except for DNS rules" k8sApiVersion= k8sNamespace=default subsys=daemon
2019-04-26T12:22:34.247898848Z level=debug msg="Starting new controller" name="sync-cnp-policy-status (v2 default/foo)" subsys=controller uuid=f868b747-681d-11e9-8c3f-080027807538
2019-04-26T12:22:34.247902267Z level=warning msg="Error parsing new CiliumNetworkPolicy rule" ciliumNetworkPolicyName=foo error="Invalid CiliumNetworkPolicy specs: L7 rules can only apply to TCP (not UDP) except for DNS rules" k8sApiVersion= k8sNamespace=default obj="TypeMeta: &TypeMeta{Kind:,APIVersion:,}, ObjectMeta: &ObjectMeta{Name:foo,GenerateName:,Namespace:default,SelfLink:/apis/cilium.io/v2/namespaces/default/ciliumnetworkpolicies/foo,UID:f867e92a-681d-11e9-a542-080027807538,ResourceVersion:10636,Generation:1,CreationTimestamp:2019-04-26 12:22:34 +0000 UTC,DeletionTimestamp:<nil>,DeletionGracePeriodSeconds:nil,Labels:map[string]string{},Annotations:map[string]string{kubectl.kubernetes.io/last-applied-configuration: {\"apiVersion\":\"cilium.io/v2\",\"kind\":\"CiliumNetworkPolicy\",\"metadata\":{\"annotations\":{},\"name\":\"foo\",\"namespace\":\"default\"},\"specs\":[{\"endpointSelector\":{\"matchLabels\":{\"any:id\":\"foo\"}},\"ingress\":[{\"fromEndpoints\":[{}],\"toPorts\":[{\"ports\":[{\"port\":\"80\",\"protocol\":\"UDP\"}],\"rules\":{\"http\":[{\"method\":\"GET\",\"path\":\"/private\"}]}}]}]}]}\n,},OwnerReferences:[],Finalizers:[],ClusterName:,Initializers:nil,ManagedFields:[],}, Specs: [&{EndpointSelector:{\"matchLabels\":{\"any:id\":\"foo\"}} Ingress:[{FromEndpoints:[{}] FromRequires:[] ToPorts:[{Ports:[{Port:80 Protocol:UDP}] Rules:0xc0019da070}] FromCIDR:[] FromCIDRSet:[] FromEntities:[] aggregatedSelectors:[]}] Egress:[] Labels:[] Description:}]Status: {map[] map[]}" subsys=k8s
2019-04-26T12:22:34.256040542Z level=debug msg="CNP status for invalid rule cannot be updated" ciliumNetworkPolicyName=foo error="Operation cannot be fulfilled on ciliumnetworkpolicies.cilium.io \"foo\": the object has been modified; please apply your changes to the latest version and try again" k8sApiVersion= k8sNamespace=default subsys=k8s
2019-04-26T12:22:34.256064628Z level=debug msg="Controller func execution time: 9.54911ms" name="sync-cnp-policy-status (v2 default/foo)" subsys=controller uuid=f868b747-681d-11e9-8c3f-080027807538
2019-04-26T12:22:34.256067833Z level=debug msg="Controller run failed" consecutiveErrors=1 error="Invalid CiliumNetworkPolicy specs: L7 rules can only apply to TCP (not UDP) except for DNS rules" name="sync-cnp-policy-status (v2 default/foo)" subsys=controller uuid=f868b747-681d-11e9-8c3f-080027807538
```

**Comment by raybejjani** [comment]
18 days later :(
https://jenkins.cilium.io/job/cilium-ginkgo/job/cilium/job/master/2873/testReport/junit/Suite-k8s-1/10/K8sServicesTest_Checks_service_across_nodes_Tests_NodePort/
